src/Common/Core/Impl/Security/Credentials.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Net;
+using System.Security;
+
+namespace Microsoft.Common.Core.Security {
+    public sealed class Credentials : ICredentials {
+        // Although NetworkCredential does support SecureString, it still exposes
```

Comment:
NetworkCredential is created and passed over rather than cached. So it won't appear in debugger.